packages/schemas/src/types/logto-config/index.ts

```diff
@@ -45,6 +51,55 @@ export const logtoOidcConfigGuard: Readonly<{
   [LogtoOidcConfigKey.CookieKeys]: oidcConfigKeyGuard.array(),
 });
 
+/**
+ * Logto JWT customizer token types, used in REST API routes.
+ */
+export enum LogtoJwtTokenKeyType {
+  AccessToken = 'access-token',
+  ClientCredentials = 'client-credentials',
+}
+
+export enum LogtoJwtTokenKey {
+  AccessToken = 'jwt.accessToken',
+  ClientCredentials = 'jwt.clientCredentials',
+}
+
+export const jwtCustomizerGuard = z
+  .object({
+    script: z.string(),
+    envVars: z.record(z.string()),
+    contextSample: jsonObjectGuard,
+    // This `tokenSample` field will be overridden by the `tokenSample` field once the `tokenType` is determined.
+    tokenSample: jsonObjectGuard,
+  })
+  .partial();
```

Comment:
No need to export this `jwtCustomizerGuard`?  Also can remove the `tokenSample`?

```ts
const sharedJwtCustomizerGuard = z.object({
    script: z.string(),
    envVars: z.record(z.string()),
    contextSample: jsonObjectGuard,
});


export const jwtCustomizerAccessTokenGuard = jwtCustomizerGuard.extend({
  // Use partial token guard since users customization may not rely on all fields.
  tokenSample: accessTokenGuard.partial().optional(),
}).partial();


export const jwtCustomizerClientCredentialsGuard = jwtCustomizerGuard.extend({
  // Use partial token guard since users customization may not rely on all fields.
  tokenSample: clientCredentialsGuard.partial().optional(),
}).partial();
```